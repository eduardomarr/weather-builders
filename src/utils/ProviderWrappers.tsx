import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { WeatherContext } from '../context/weather';
import theme from '../styles/theme';

type WrapperProps = {
  children: React.ReactNode;
};

const mockedContextValues = {
  weather: {
    description: 'description',
    temp: 10,
    feelsLike: 16,
    tempMin: 10,
    tempMax: 16,
    pressure: 10,
    humidity: 50,
    windSpeed: 50,
    district: 'SÃ£o Paulo',
    visibility: 'visibility',
    sunset: 'sunset',
    sunrise: 'sunrise',
    icon: 'icon',
  },

  getCurrentWeather: jest.fn(),
  loading: false,
  setLoading: jest.fn(),
};

export const ThemeWrapper: React.FC<WrapperProps> = ({ children }) => {
  return (
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
  );
};

export const ContextWrapper: React.FC<WrapperProps> = ({ children }) => (
  <WeatherContext.Provider value={mockedContextValues}>
    {children}
  </WeatherContext.Provider>
);
