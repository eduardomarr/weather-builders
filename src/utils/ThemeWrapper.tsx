import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import theme from '../styles/theme';

type WrapperProps = {
  children: React.ReactNode;
};

export const ThemeWrapper: React.FC<WrapperProps> = ({ children }) => {
  return (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
};
