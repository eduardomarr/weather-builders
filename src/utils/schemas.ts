import * as Yup from 'yup';

export const signInSchema = Yup.object().shape({
  email: Yup.string().required('E-mail é obrigatório').email('E-mail Inválido!'),
  password: Yup.string().required('Senha é obrigatório'),
});
