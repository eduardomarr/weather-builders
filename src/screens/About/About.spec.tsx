import React from 'react';
import { ThemeWrapper } from '../../utils/ProviderWrappers';

import { About } from '.';

import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { render } from '@testing-library/react-native';

describe('About Screen', () => {

  it('should be render correctly', () => {

    const { getByTestId } = render(
      <About />,
      {wrapper: ThemeWrapper}
    );

    expect(getByTestId('container')).toBeOnTheScreen();

  });

  it('should be styled correctly', () => {
    const tree = renderer.create(
    <ThemeWrapper>
      <About />
    </ThemeWrapper>
    ).toJSON() as ReactTestRendererJSON;

      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyle({
        backgroundColor: '#090932',
    });
  });

  it('should have Logo Component', () => {

      const { getByTestId } = render(
        <About />,
        {wrapper: ThemeWrapper}
      );

      expect(getByTestId('logo-component')).toBeOnTheScreen();

    });

    it('should have Current Version Component', () => {

      const { getByTestId } = render(
        <About />,
        {wrapper: ThemeWrapper}
      );

      expect(getByTestId('logo-component')).toBeOnTheScreen();

    });

  });
