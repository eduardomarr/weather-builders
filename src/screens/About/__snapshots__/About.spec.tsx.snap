// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Screen should be styled correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#090932",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "paddingBottom": 80,
        "paddingLeft": 32,
        "paddingRight": 32,
        "paddingTop": 80,
      },
    ]
  }
  testID="container"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "paddingBottom": 24,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 24,
          },
        ]
      }
      testID="logo-component"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/logo.png",
          }
        }
        style={
          Array [
            Object {
              "height": 294,
              "marginBottom": 16,
              "width": 294,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Manrope-Bold",
              "fontSize": 47,
            },
          ]
        }
      >
        Weather Builders
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Manrope-Medium",
            "fontSize": 35,
            "marginBottom": 32,
          },
        ]
      }
      testID="version-component"
    >
      Vers√£o: 1.0.0
    </Text>
  </View>
</View>
`;
